Single-build handbook with relocated solutions (#30)
neuroinformatics-unit/course-animals-in-motion: CONTRIBUTING.md

```bash
quarto render book
```

Code cells in `.qmd` files execute automatically on render.
Their results are cached (`execute: cache: true` in `book/_quarto.yml`),

> so on subsequent renders only cells you've edited re-run.

Reviewer: ```suggestion
so on subsequent renders only cells you've modified are re-run.
```

If you need to force a full re-execution—e.g. after changing data or an

> imported dependency that Quarto can't see—refresh the cache:

Reviewer: ```suggestion
imported dependency that Quarto cannot detect—refresh the cache:
```

In general, [cross-referencing objects](https://quarto.org/docs/manuscripts/authoring/vscode.html#cross-ref) (e.g. figures, tables, chapters, equations, citations, etc.) should be done using the `@ref` syntax, e.g. `See @fig-overview for more details`.

### Adding exercises and their solutions

Exercises and their solutions are authored inline in the chapter source, but at
render time each solution is moved into a per-chapter "Solutions" section by the

> `book/collect-solutions.lua` filter (registered in `book/_quarto.yml`).

Reviewer: ```suggestion
`book/collect-solutions.lua` [filter](https://quarto.org/docs/extensions/filters.html) (registered in `book/_quarto.yml`).
```

We use [Calendar Versioning (CalVer)](https://calver.org/) and specifically the `YYYY.0M` scheme (e.g. `2025.08` for August 2025).

To create a new release, first update the `book/index.qmd` file. Specifically, add a row like the following to the "Versions" table:

```md

> | `v2026.08` | version used for the OSSS in August 2025 |

Reviewer: ```suggestion
| `v2026.08` | version used for the OSSS in August 2026 |
```

```

> You also need to create a new tag in the `vYYYY.0M` format (e.g. `v2025.08`)

Reviewer: ```suggestion
You also need to create a new tag in the `vYYYY.MM` format (e.g. `v2025.08`)
```

- Pushes to the `main` branch
- Pull requests
- Releases, i.e. tags starting with `v` (e.g., `v2026.08`)
- Manual dispatches

The workflow is built using [GitHub actions](https://docs.github.com/en/actions) and includes three jobs:

- **linting**: running the [pre-commit hooks](#pre-commit-hooks);
- **build**: rendering the Quarto book and uploading the rendered artifact;
- **deploy**: deploying the book artifact(s) to the `gh-pages` branch (only for pushes to the `main` branch and releases).

Each release version is deployed to a folder in the `gh-pages` branch, with the same name as the release tag (e.g., `v2026.08`).

There's also a special folder called `dev` that is deployed for pushes to the `main` branch.

> Versions up to and including `v2025.10` were additionally deployed to a `vYYYY.0M-answers` folder (separate builds for with and without solutions). Those folders remain online as historical archives; new versions no longer produce them.

Reviewer: ```suggestion
Versions up to and including `v2025.10` were additionally deployed to a `vYYYY.MM-answers` folder (separate builds for with and without solutions). Those folders remain online as historical archives; new versions no longer produce them.
```
